pycbc/workflow/minifollowups.py

```diff
@@ -852,3 +1019,222 @@ def make_skipped_html(workflow, skipped_data, out_dir, tags):
+        '.png',
+        '-o',
+    )
+    workflow += skymap_plot_node
+
+    all_output_files = xml_node.output_files + bayestar_node.output_files + \
+        skymap_plot_node.output_files
+    return all_output_files
+
+
+def setup_upload_prep_minifollowups(workflow, coinc_file, xml_all_file,
+                                    single_triggers, psd_files,
+                                    tmpltbank_file, insp_segs, insp_data_name,
+                                    insp_anal_name, dax_output, out_dir,
+                                    tags=None):
+    """ Create plots that followup the Nth loudest coincident injection
+    from a statmap produced HDF file.
+
+    Parameters
+    ----------
+    workflow: pycbc.workflow.Workflow
+        The core workflow instance we are populating
+    coinc_file:
+    single_triggers: list of pycbc.workflow.File
+        A list cointaining the file objects associated with the merged
+        single detector trigger files for each ifo.
+    psd_files: list of pycbc.workflow.File
+        A list containing the file objects associated with the merged
+        psd files for each ifo.
+    xml_all_file : workflow file object
+        XML File containing all foreground events
+    tmpltbank_file: pycbc.workflow.File
+        The file object pointing to the HDF format template bank
+    insp_segs: SegFile
+       The segment file containing the data read and analyzed by each inspiral
+       job.
+       The segment file containing the data read and analyzed by each inspiral
+       job.
+    insp_data_name: str
+        The name of the segmentlist storing data read.
+    insp_anal_name: str
+        The name of the segmentlist storing data analyzed.
+    dax_output : directory
+        Location of the dax outputs
+    out_dir: path
+        The directory to store minifollowups result plots and files
+    tags: {None, optional}
+        Tags to add to the minifollowups executables
+
+    Returns
+    -------
+    layout: list
+        A list of tuples which specify the displayed file layout for the
+        minifollowups plots.
+    """
+    logging.info('Entering minifollowups module')
+
+    if not workflow.cp.has_section('workflow-minifollowups'):
+        logging.info('There is no [workflow-minifollowups] section in configuration file')
+        logging.info('Leaving minifollowups')
+        return
+
+    tags = [] if tags is None else tags
+    makedir(dax_output)
+    makedir(out_dir)
+
+    # turn the config file into a File class
+    config_path = os.path.abspath(dax_output + '/' + '_'.join(tags) + \
+                                  'upload_prep_minifollowup.ini')
+    workflow.cp.write(open(config_path, 'w'))
+
+    config_file = resolve_url_to_file(config_path)
+
+    exe = Executable(workflow.cp, 'upload_prep_minifollowup',
+                     ifos=workflow.ifos, out_dir=dax_output, tags=tags)
+
+    node = exe.create_node()
+    node.add_input_opt('--config-files', config_file)
```

Comment:
If you're doing this (which is right) you *must not* use `resolve_url` in the `pycbc_upload_prep_minifollowup`. If you do you'll end up with file paths pointing to some local condor filesystem ... and this will fail. 

I tried to point out a good example for this, but I think I found broken code (https://github.com/gwastro/pycbc/blob/master/bin/workflows/pycbc_make_sbank_workflow#L205). That example is a workflow that could be a top-level workflow or a subworkflow. The seed bank would be passed from the top-level workflow *or* is an input file. I think this would fail if seed-file is given and it is not a subworkflow.

The heart of the issue is that `resolve_url` does this https://github.com/gwastro/pycbc/blob/master/pycbc/workflow/core.py#L2120, and `from_file` does not. We *do* need to add a PFN if this is a top-level workflow and this is a global input, we *must not* add the PFN if it is not (especially because in this case we cannot predict where the statmap-file and bank-file would actually be).

Reply:
Should I add back in the `--is-subworkflow` option for this then?

Reply:
I'm unsure of this now, this seems to be working in my latest tests (using the `copy_to_cwd=True` argument), however my testing uses a cache file, so the files for which resolve_url is called are definitely not still in local. Could this be an issue?

This also matches the way this is called in the foreground and injection minifollowups - will they need a fix as well? I'm surprised we haven't seen issues before in that case

Reply:
I think the reuse cache *is* affecting this test. There's no file management going on between the workflows when everything just comes as a pregenerated input file in a guaranteed location. Is there any way to turn this off, perhaps let the workflow make `H1L1-PAGE_FOREGROUND_XMLALL-XXX.xml` again from the already existing inputs?

I do see the "incorrect" file location in the dax file:

```
  - lfn: H1L1-INSP_SEGMENTS-1368975466-1121610.xml
    pfns:
    - site: local
      pfn: file:///local/condor/execute/dir_2994154/pegasus.kxIWl4525/H1L1-INSP_SEGMENTS-1368975466-1121610.xml
```

but it's possible that pegasus will always prefer PFN coming from reuse caches .... And it uses the reuse cache functionality to send intermediate files up and down. So this may work okay without changes ...

Reply:
Any update on what happens without using a reuse cache (or a workflow I can look at with this in place).